base/src/pretty_print.rs

```diff
@@ -264,14 +264,52 @@ impl<'a> ExprPrinter<'a> {
     where
         Id: AsRef<str>,
     {
-        self.pretty_expr_(BytePos::from(0), expr).append(
+        self.pretty_expr_with_shebang_line(BytePos::from(0), expr).append(
             self.comments(Span::new(
                 expr.span.end,
                 BytePos::from(self.source.src().len()),
             )),
         )
     }
 
+    fn find_shebang_line(&'a self) -> Option<&str> {
+        let src = self.source.src();
+        let len = src.len();
+        if len > 1 && &src[0..2] == "#!" {
```

Comment:
No idea why I didn't go with this in the first place.  This is a whole lot cleaner.  I think my brain was assuming rust didn't have these methods built in.  VSCode autocomplete seems to break a lot so I probably should look over the APIs in the docs more.  Thanks for the tips.